src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -194,52 +194,96 @@ public void importEntries(List<BibEntry> entries) {
     }
 
     public void importEntryWithDuplicateCheck(BibDatabaseContext bibDatabaseContext, BibEntry entry) {
-        ImportCleanup cleanup = new ImportCleanup(bibDatabaseContext.getMode());
-        BibEntry cleanedEntry = cleanup.doPostCleanup(entry);
+        BibEntry cleanedEntry = cleanUpEntry(bibDatabaseContext, entry);
+
+        Optional<BibEntry> existingDuplicateInLibrary = findDuplicate(bibDatabaseContext, cleanedEntry);
         BibEntry entryToInsert = cleanedEntry;
 
-        Optional<BibEntry> existingDuplicateInLibrary = new DuplicateCheck(Globals.entryTypesManager).containsDuplicate(bibDatabaseContext.getDatabase(), entryToInsert, bibDatabaseContext.getMode());
         if (existingDuplicateInLibrary.isPresent()) {
-            DuplicateResolverDialog dialog = new DuplicateResolverDialog(existingDuplicateInLibrary.get(), entryToInsert, DuplicateResolverDialog.DuplicateResolverType.IMPORT_CHECK, bibDatabaseContext, stateManager, dialogService, preferencesService);
-            switch (dialogService.showCustomDialogAndWait(dialog).orElse(DuplicateResolverDialog.DuplicateResolverResult.BREAK)) {
-                case KEEP_RIGHT:
-                    bibDatabaseContext.getDatabase().removeEntry(existingDuplicateInLibrary.get());
-                    break;
-                case KEEP_BOTH:
-                    break;
-                case KEEP_MERGE:
-                    bibDatabaseContext.getDatabase().removeEntry(existingDuplicateInLibrary.get());
-                    entryToInsert = dialog.getMergedEntry();
-                    break;
-                case KEEP_LEFT:
-                case AUTOREMOVE_EXACT:
-                case BREAK:
-                default:
-                   return;
+            entryToInsert = handleDuplicates(bibDatabaseContext, cleanedEntry, existingDuplicateInLibrary.get());
+            // If handleDuplicates returned null, stop processing
+            if (entryToInsert == null) {
+                return;
             }
         }
-        // Regenerate CiteKey of imported BibEntry
+
+        regenerateCiteKey(entryToInsert);
+        bibDatabaseContext.getDatabase().insertEntry(entryToInsert);
+
+        setAutomaticFieldsForEntry(entryToInsert);
+
+        addEntryToGroups(entry);
```

Comment:
I think, the new entry should be inserted, not the original one?. Maybe, rename `entry` to `originalEntry`

```suggestion
        addEntryToGroups(entryToInsert);
```